FingerprintDialogKotlin/app/src/main/java/com/example/android/fingerprintdialog/MainActivity.kt

```diff
@@ -287,6 +280,53 @@ class MainActivity : AppCompatActivity(),
         return super.onOptionsItemSelected(item)
     }
 
+    private fun instanceOfBiometricPrompt(): BiometricPrompt {
+        val executor = ContextCompat.getMainExecutor(this)
+
+        val callback = object : BiometricPrompt.AuthenticationCallback() {
+            override fun onAuthenticationError(errorCode: Int, errString: CharSequence) {
+                super.onAuthenticationError(errorCode, errString)
+                Log.d(TAG, "$errorCode :: $errString")
+                if(errorCode == BiometricPrompt.ERROR_NEGATIVE_BUTTON) {
+                    loginWithPassword() // because negative button says use application password
+                }
+            }
+
+            override fun onAuthenticationFailed() {
+                super.onAuthenticationFailed()
+                Log.d(TAG, "Authentication failed for an unknown reason")
+            }
+
+            override fun onAuthenticationSucceeded(result: BiometricPrompt.AuthenticationResult) {
+                super.onAuthenticationSucceeded(result)
+                Log.d(TAG, "Authentication was successful")
+                onPurchased(true,result.cryptoObject)
+            }
+        }
+
+        val biometricPrompt = BiometricPrompt(this, executor, callback)
+        return biometricPrompt
+    }
+
+    private fun getPromptInfo(): BiometricPrompt.PromptInfo {
```

Comment:
nit: This is creating the prompt each time, not just getting it. Maybe rename to `createPromptInfo`?